implants/eldritch/src/pivot/port_scan_impl.rs

```diff
@@ -489,21 +439,42 @@ mod tests {
         Ok(())
     }
 
+    async fn local_bind_tcp() -> TcpListener {
+        // Try three times to bind to a port
+        let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
+        return listener;
+    }
+
+    async fn local_accept_tcp(listener: TcpListener) -> Result<()> {
+        // Accept new connection
+        let (mut socket, _) = listener.accept().await?;
+        // Split reader and writer references
+        let (mut reader, mut writer) = socket.split();
+        // Copy from reader to writer to echo message back.
+        let bytes_copied = copy(&mut reader, &mut writer).await?;
+        // If message sent break loop
+        if bytes_copied > 1 {
+            return Ok(());
+        } else {
+            return Err(anyhow::anyhow!("Failed to copy any bytes"));
+        }
+    }
+
     #[tokio::test]
     async fn test_portscan_tcp() -> anyhow::Result<()> {
-        let test_ports =  allocate_localhost_unused_ports(4, "tcp".to_string()).await?;
-
+        // Allocate unused ports
+        let mut bound_listeners_vec: Vec<TcpListener> = vec![];
+        for _ in 0..(3) {
```

Comment:
![image](https://user-images.githubusercontent.com/16250309/223310309-7a95d91c-59b4-4bba-8073-2eb9e8f06f3a.png)

Let's define this as a constant